### PR TITLE
Flink job endpoints

### PR DESCRIPTION
**Pull request by janelletavares**
## Description of change

Supports the CLI work to describe, list, and deploy Flink Jobs

Fixes https://github.com/meroxa/product/issues/778

## Type of change

- [x]  New feature
- [ ]  Bug fix
- [ ]  Refactor
- [ ]  Documentation

## How was this tested?

- [x]  Unit Tests
- [ ]  Tested in staging
- [ ]  Tested in minikube

## Demo

**Before:**

**After:**

### CODE REVIEW

File: pkg/meroxa/source.go [hunk 1]
```diff
@@ -16,6 +16,7 @@ type Source struct {
 }
 
 type CreateSourceInputV2 struct {
```
[the remark this answers is not in the document]
Comment (janelletavares) [reply pkg/meroxa/source.go:18]: no, turbine also uses this endpoint

File: pkg/meroxa/flink_job.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+const flinkJobsBasePath = "/v1/flink-jobs"
+
+type FlinkJobState string
+
+const (
+	FlinkJobStateCancelled    FlinkJobState = "cancelled"
+	FlinkJobStateCancelling   FlinkJobState = "cancelling"
+	FlinkJobStateCreated      FlinkJobState = "created"
+	FlinkJobStateFailed       FlinkJobState = "failed"
```
Comment (janelletavares) [inline pkg/meroxa/flink_job.go:19]:
```suggestion
	FlinkJobStateCreated      FlinkJobState = "created"
	FlinkJobStateDOA      FlinkJobState = "doa"
	FlinkJobStateFailed       FlinkJobState = "failed"
```

File: pkg/meroxa/source.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jayjayjpg) [reply pkg/meroxa/source.go:18]: Thank you!

File: pkg/meroxa/flink_job.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+	var fj *FlinkJob
+	if err = json.NewDecoder(resp.Body).Decode(&fj); err != nil {
+		return nil, err
+	}
+
+	return fj, nil
+}
+
+func (c *client) DeleteFlinkJob(ctx context.Context, nameOrUUID string) error {
```
Comment (janelletavares) [inline pkg/meroxa/flink_job.go:105]: sweet

**Review by ericcheatham — APPROVED** [review]

**Pull request merged**